traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/models/Traveler.java

```diff
@@ -532,10 +525,38 @@ protected void main() {
         TaskManager.getMainTaskManager().addTask(fetchBlockTask);
     }
 
+    public static void fetchPurchaseForm(PartnerOfferingProduct product, List<ProductOffering> passes, FetchPurchaseFormCallback fetchPurchaseFormCallback) {
+        if (!isInitialized()) return;
+
+        AuthenticatedUrlRequest request;
+
+        request = Router.partnerOfferingQuestions(localInstance.session, product, localInstance.applicationContext);
+
+        AuthenticatedRemoteNetworkRequestTask<List<QuestionGroup>> fetchPurchaseFormTask =
+                new AuthenticatedRemoteNetworkRequestTask<>(localInstance.session, localInstance.applicationContext,
+                        request, new ArrayMappingFactory<>(new QuestionGroup.QuestionGroupObjectMappingFactory()));
+
+        BlockTask fetchBlockTask = new BlockTask() {
+            @Override
+            protected void main() {
+                if (null != fetchPurchaseFormTask.getError()) {
+                    fetchPurchaseFormCallback.onPurchaseFormFetchError(fetchPurchaseFormTask.getError());
+                    Log.e(TAG, fetchPurchaseFormTask.getError().getMessage());
+                } else {
+                    fetchPurchaseFormCallback.onPurchaseFormFetchSuccess(new PurchaseForm(product, passes, fetchPurchaseFormTask.getResource()));
+                }
+            }
+        };
+
+        fetchBlockTask.addDependency(fetchPurchaseFormTask);
+        localInstance.taskManager.addTask(fetchPurchaseFormTask);
+        TaskManager.getMainTaskManager().addTask(fetchBlockTask);
+    }
+
```

Comment:
Overall, this whole purchaseForm work does not belong in this PR. And if it did, it should be reflected in your commit message. House rule says that commits shouldn't do more than what the message commit states. They also shouldn't do less. At this point, this commit is doing more than what it says (because it is not only changing parsing methods it is also splitting purchase form). 

And if you're going to start some refactoring work like this, finish it, otherwise don't start it and leave it for another PR. This is messy.